### DISCUSSION TITLE
IsLoading delay

### DISCUSSION THREAD

**Discussion by mweel1** [TanStack/query#5938] [General]
I am using the isLoading property to disable the button so users dont double press it.

However, the isLoading has a delay, most likely not changing state until the network actually starts making the request.  I can actually double press so its not working.

Obviously, I dont want to introduce another state variable for this very common use case.

Is there anything else i could use from the hooks, or would it make sense to move that isLoading state higher in this component?

This seems like a very common use case, and introducing another state variable just doesn't seem like the way.

**Comment by TkDodo** [comment]
can you show a codesandbox reproduction ?

**Comment by Birkbjo** [comment]
Are you sure you're not `await`-ing something before you call mutate? I stumbled upon this recently when I used `fetchQuery` instead of `getQueryData` when grabbing the cached data for the mutation.
